### PR TITLE
Add filter permissions check

### PR DESCRIPTION
**Pull request by bond95**
Fixes: https://github.com/oVirt/ovirt-web-ui/issues/118

### CODE REVIEW

**Comment by ovirt-infra** [comment]
The commiter is not a member of organization nor whitelist. Please contact maintainer for patch verification.

**Comment by ovirt-infra** [comment]
The commiter is not a member of organization nor whitelist. Please contact maintainer for patch verification.

**Comment by ovirt-infra** [comment]
The commiter is not a member of organization nor whitelist. Please contact maintainer for patch verification.

**Comment by ovirt-infra** [comment]
The commiter is not a member of organization nor whitelist. Please contact maintainer for patch verification.

**Comment by ovirt-infra** [comment]
The commiter is not a member of organization nor whitelist. Please contact maintainer for patch verification.

**Comment by ovirt-infra** [comment]
The commiter is not a member of organization nor whitelist. Please contact maintainer for patch verification.

**Comment by ovirt-infra** [comment]
The commiter is not a member of organization nor whitelist. Please contact maintainer for patch verification.

**Comment by ovirt-infra** [comment]
The commiter is not a member of organization nor whitelist. Please contact maintainer for patch verification.

**Comment by ovirt-infra** [comment]
The commiter is not a member of organization nor whitelist. Please contact maintainer for patch verification.

**Comment by ovirt-infra** [comment]
The commiter is not a member of organization nor whitelist. Please contact maintainer for patch verification.

**Comment by ovirt-infra** [comment]
The commiter is not a member of organization nor whitelist. Please contact maintainer for patch verification.

**Comment by ovirt-infra** [comment]
The commiter is not a member of organization nor whitelist. Please contact maintainer for patch verification.

**Comment by jelkosz** [comment]
@mareklibra maybe it would be good to have also an option to disable this if the user decides. E.g. if the admin decides, he will see only the filtered data.

**Comment by mareklibra** [comment]
@jelkosz , agree

File: src/sagas.js [hunk 1]
```diff
@@ -95,7 +98,8 @@ function* login (action) {
     const username = action.payload.credentials.username
     // persistTokenToSessionStorage({ token, username })
     yield put(loginSuccessful({ token, username }))
-
+    yield put(checkFilterPermissions())
+    yield delay(1000)
```
Comment (mareklibra) [inline src/sagas.js]: Why is this needed?

File: src/sagas.js [hunk 2]
```diff
@@ -372,6 +376,11 @@ function* schedulerPerMinute (action) {
   }
 }
 
+function* fetchPermissionWithoutFilter (action) {
+  const data = yield callExternalAction('checkFilter', Api.checkFilter, { action: 'CHECK_FILTER' })
+  yield put(setUserFilterPermission(data.error !== undefined))
```
Comment (mareklibra) [inline src/sagas.js]: @jelkosz , is this how the API is intended to be used for this kind of information?

**Comment by bond95** [comment]
@mareklibra resolved conflicts, please check

File: src/ovirtapi.js [hunk 3]
```diff
@@ -15,7 +15,7 @@ OvirtApi = {
       throw new Exception(`OvirtApi in '${methodName}': missing login`)
     }
   },
-  _httpGet ({ url, custHeaders = { 'Accept': 'application/json', Filter: true } }) {
+  _httpGet ({ url, custHeaders = { 'Accept': 'application/json', Filter: Selectors.getFilter() } }) {
```
Comment (mareklibra) [inline src/ovirtapi.js]: Please update OvirtApi.console() as well

File: src/sagas.js [hunk 4]
```diff
@@ -106,13 +112,12 @@ function* login (action) {
...
     // persistTokenToSessionStorage({ token, username })
     yield put(loginSuccessful({ token, username }))
-
     const oVirtMeta = yield callExternalAction('getOvirtApiMeta', Api.getOvirtApiMeta, action)
     if (!oVirtMeta['product_info']) { // REST API call failed
       yield put(yield put(loadInProgress({ value: false })))
     } else {
       if (yield checkOvirtApiVersion(oVirtMeta)) {
-        yield put(getAllVms({ shallowFetch: false }))
+        yield put(checkFilterPermissions())
```
Comment (mareklibra) [inline src/sagas.js]: can we avoid this nesting of getAllVms() call?

File: src/sagas.js [hunk 5]
```diff
@@ -389,6 +394,17 @@ function* schedulerPerMinute (action) {
...
 
+function* fetchPermissionWithoutFilter (action) {
+  const data = yield callExternalAction('checkFilter', Api.checkFilter, { action: 'CHECK_FILTER' })
+  yield put(changeFilterPermissions(data.error !== undefined))
+  yield put(setAdministrator(data.error === undefined))
+}
+
+function* changeUserFilterPermission (action) {
+  yield put(setUserFilterPermission(action.payload.filter))
+  yield put(getAllVms({ shallowFetch: false }))
```
Comment (mareklibra) [inline src/sagas.js]: By this change, the flow becomes less understandable. Can it be moved back?

File: src/sagas.js (hunk 4 above)
Comment (bond95) [inline src/sagas.js]: I tried do it with delay, but for me it more worst solution than nesting

File: src/sagas.js (hunk 5 above)
Comment (bond95) [inline src/sagas.js]: As I wrote above, getAllVms must be called only after it check user role, for me there only two ways, first make delay (which is the worst). and second is make it nested. __changeUserFilterPermission__ is only use in logining and changing filter, so I think make flow little more complex for it, is not so bad solution

File: src/ovirtapi.js (hunk 3 above)
Comment (mareklibra) [inline src/ovirtapi.js]: Still missing in latest commit, please don't forget it
